libr/bin/format/dmp/dmp64.c

```diff
@@ -0,0 +1,236 @@
+/* radare2 - LGPL - Copyright 2020 - abcSup */
+
+#include <r_types.h>
+#include <r_util.h>
+
+#include "dmp64.h"
+
+static int r_bin_dmp64_init_memory_runs(struct r_bin_dmp64_obj_t *obj) {
+	int i, j;
+	dmp64_p_memory_desc *mem_desc = &obj->header->PhysicalMemoryBlockBuffer;
+	if (!memcmp (mem_desc, DMP_UNUSED_MAGIC, 4)) {
+		eprintf ("Warning: Invalid PhysicalMemoryDescriptor\n");
+		return false;
+	}
+	ut64 num_runs = mem_desc->NumberOfRuns;
+	if (num_runs * sizeof (dmp_p_memory_run) >= 0x348) {
```

Comment:
what is this magic number?

Reply:
Probably you should DEFINE it somewhere with a meaningful name.

Reply:
That is the offset to the next property. That line is to check if the PhysicalMemoryBlockBuffer has valid size. Is there a way to easily retrieve a specific offset of a item in a structure without DEFINE?
```
0x80  : KdDebuggerDataBlock            ['unsigned long long']
0x88  : PhysicalMemoryBlockBuffer      ['_PHYSICAL_MEMORY_DESCRIPTOR']
0x348 : ContextRecord                  ['array', 3000, ['unsigned char']]
0xf00 : Exception                      ['_EXCEPTION_RECORD64']
```

Reply:
DEFINE would be fine, just wondering if there is another way without doing something like, `&(obj->header->NumberOfPages) - obj->header`

Reply:
Nvm, I found r_offsetof!